virtinst/osdict.py

```diff
@@ -16,6 +16,32 @@
 from .logger import log
 
 
+def _process_medialist(medialist):
+    nmedias = medialist.get_length()
+    osids = []
+
+    if nmedias == 0:
+        return None
+    elif nmedias == 1:
+        return medialist.get_nth(0)
+    else:
+        # If we can't determine a single definitive match, print the
+        # possible matches and return None
+        for n in range(nmedias):
+            # some isos can return multiple matches of the same OS with
+            # different architectures. Ignore duplicates.
+            id = medialist.get_nth(n).get_os().get_short_id()
```

Comment:
please use the OS ID rather than the (first) short-id: strictly speaking, libosinfo does not reject the same short-id in different OSes (a test for this was added few months ago in osinfo-db)